src/lib/storage/proposal/partition_creator.rb

```diff
@@ -211,17 +217,27 @@ def create_non_lvm_complex
 
         # Creates a partition and the corresponding filesystem for each volume
         #
+        # First it tries to create the partitions for the volumes that need to
+        # be in a particular disk.
+        # @raise an error if that's not possible.
+        #
         # It tries to honor the value of #max_start_offset for each volume, but
         # it does not raise an exception if that particular requirement is
         # impossible to fulfill, since it's usually more a recommendation than a
         # hard limit.
         #
         # @param volumes [Array<PlannedVolume>]
         def create_volumes_partitions(volumes)
-          volumes.sort_by_attr(:max_start_offset).each do |vol|
+          volumes.sort_by_attr(:disk, :max_start_offset).each do |vol|
```

Comment:
Well, now that we can sort by several attributes we could do something like this if you want:
```ruby
# :disk, and :max_start_offset because are relevant for the algorithm,
# :mount_point and :filesystem_type just to ensure consistency of order
volumes.sort_by_attr(:disk, :max_start_offset, :mount_point, :filesystem_type).each do |vol|
```